demos/common/cpp/utils/include/utils/config_factory.h

```diff
@@ -16,11 +16,13 @@
 
 #pragma once
 #include <map>
+#include <set>
 #include <string>
 #include "gflags/gflags.h"
 
 struct CnnConfig {
-    std::string devices;
+    std::string deviceName;
+    std::set<std::string> devices;
```

Comment:
we'd better put devices to private or protected section with public getter - to avoid setting devices explicetely

Reply:
Ok, made it as protected.